src/SFA.DAS.ApprenticeCommitments/Data/Models/Apprentice.cs

```diff
@@ -1,17 +1,52 @@
 ﻿using System;
+using System.Collections.Generic;
 using System.ComponentModel.DataAnnotations.Schema;
+using System.Net.Mail;
 
 namespace SFA.DAS.ApprenticeCommitments.Data.Models
 {
-    [Table("Apprentice")]
     public class Apprentice
     {
-        public long Id { get; set; }
-        public string FirstName { get; set; }
-        public string LastName { get; set; }
-        public Guid UserIdentityId { get; set; }
-        public string Email { get; set; }
-        public DateTime DateOfBirth { get; set; }
-        public DateTime CreatedOn { get; private set; }
+        private Apprentice()
+        {
+        }
+
+        public Apprentice(string firstName, string lastName, Guid userIdentityId, MailAddress email, DateTime dateOfBirth)
+        {
+            FirstName = firstName;
+            LastName = lastName;
+            UserIdentityId = userIdentityId;
+            Email = email;
+            DateOfBirth = dateOfBirth;
+            PreviousEmails = new[] { new ApprenticeEmailHistory(email) };
+        }
+
+        public long Id { get; private set; }
+        public string FirstName { get; private set; }
+        public string LastName { get; private set; }
+        public Guid UserIdentityId { get; private set; }
+        public MailAddress Email { get; private set; }
+        public ICollection<ApprenticeEmailHistory> PreviousEmails { get; private set; }
+        public DateTime DateOfBirth { get; private set; }
+        public DateTime CreatedOn { get; private set; } = DateTime.UtcNow;
+
```

Comment:
Just a mater of interest here, what happens with EF when it reads an existing Apprentice record (how's it set the Id)? Is it done by reflection? I know Newtonsoft reads constructors and guesses the parameter names relate.

Same goes for CreateOn field

Reply:
EF will set all the writeable properties it can find, even the private ones.  Actually, it can now also set read-only properties, if we make some changes to the mapping in the context - see https://docs.microsoft.com/en-us/ef/core/modeling/constructors.  

EF is capable of using the parameterised constructor to create the object, as long as the parameter names are (case-insensitive) the same as the property names.  However in this particular case the constructor is doing extra work that we don't want to happen when just hydrating the object from the db - namely adding the original email address to the PreviousEmailAddresses list.  So I provide a private default constructor that EF will use by preference.